src/features/community/useCommunity.tsx

```diff
@@ -0,0 +1,43 @@
+import { useAppDispatch, useAppSelector } from "../../store";
+import { useEffect } from "react";
+import { getCommunity } from "./communitySlice";
+import useCommunityActions from "./useCommunityActions";
+
+// fetches the community info (subscribed, etc) from the server if its not available in local
+export default function useCommunity(communityHandle: string) {
+  const dispatch = useAppDispatch();
+
+  const communityByHandle = useAppSelector(
+    (state) => state.community.communityByHandle,
+  );
+
+  useEffect(() => {
+    if (communityByHandle[communityHandle]) return;
+
+    dispatch(getCommunity(communityHandle));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [communityHandle]);
+
+  const {
+    isFavorite,
+    isSubscribed,
+    isBlocked,
+    subscribe,
+    favorite,
+    block,
+    sidebar,
+    view,
+  } = useCommunityActions(communityByHandle[communityHandle]!);
```

Comment:
not sure how to handle this case if `communityByHandle[communityHandle]` is undefined. any suggestions? I've not the hook `useCommunity` anywhere for now though